combine/combine.go

```diff
@@ -0,0 +1,514 @@
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// findFilePathsInDir finds files in the given dir that have basenames with the
+// given suffix. Dies on error.
+func findFilePathsInDir(dir, suffix string) ([]string, error) {
+	paths, err := filepath.Glob(fmt.Sprintf("%s/*%s", dir, suffix))
+	if err != nil || len(paths) == 0 {
+		return paths, Error("Could not find path")
+	}
+
+	return paths, nil
+}
+
+// createOutputFileInDir creates a file for writing in the given dir with the
+// given basename. Dies on error.
+func createOutputFileInDir(dir, basename string) (*os.File, error) {
+	file, err := os.Create(filepath.Join(dir, basename))
+	if err != nil {
+		return file, err
+	}
+
+	return file, nil
+}
+
+// mergeStreamToOutputFunc is one of our merge*StreamTo* functions.
+type mergeStreamToOutputFunc func(data io.ReadCloser, output *os.File) error
+
+// mergeFilesAndStreamToOutput merges the inputs files and streams the content
+// to the streamFunc.
+func mergeFilesAndStreamToOutput(inputs []string, output *os.File, streamFunc mergeStreamToOutputFunc) error {
+	sortMergeOutput, cleanup, err := mergeSortedFiles(inputs)
+	if err != nil {
+		return err
+	}
+
+	if err = streamFunc(sortMergeOutput, output); err != nil {
+		return err
+	}
+
+	if err = cleanup(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// mergeSortedFiles shells out to `sort -m` to merge pre-sorted files together.
+// Returns a pipe of the output from sort, and function you should call after
+// you've finished reading the output to cleanup.
+func mergeSortedFiles(inputs []string) (io.ReadCloser, func() error, error) {
+	cmd := exec.Command("sort", "-m", "--files0-from", "-")
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "LC_ALL=C")
+
+	sortStdin, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sortMergeOutput, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if err = cmd.Start(); err != nil {
+		return nil, nil, err
+	}
+
+	if err = sendFilePathsToSort(sortStdin, inputs); err != nil {
+		return nil, nil, err
+	}
+
+	return sortMergeOutput, cmd.Wait, nil
+}
+
+// sendFilePathsToSort will pipe the given paths null terminated to the pipe.
+// For use with the StdinPipe of an exec.Command for `sort -m --files0-from -`.
+// The in is closed afterwards.
+func sendFilePathsToSort(in io.WriteCloser, paths []string) error {
+	for _, path := range paths {
+		if _, err := in.Write([]byte(path + string(rune(0)))); err != nil {
+			return err
+		}
+	}
+
+	return in.Close()
+}
+
+// openFiles opens the given files for reading.
+func OpenFiles(paths []string) ([]*os.File, error) {
+	files := make([]*os.File, len(paths))
+
+	for i, path := range paths {
+		file, err := os.Open(path)
+		if err != nil {
+			return files, err
+		}
+
+		files[i] = file
+	}
+
+	return files, nil
+}
+
+// THIS FUNCTION NEEDS TO BE GENERALISED TO ALL FILES, NOT EXCLUSIVELY PGZIP.
```

Comment:
Just change the type of 'output' to the type that CopyBuffer takes as first arg.